src/wayland/shell/xdg/mod.rs

```diff
@@ -1115,6 +1123,30 @@ impl ToplevelSurface {
             .unwrap(),
         )
     }
+
+    /// Returns true if the provided surface is the parent of this toplevel surface.
+    pub fn is_parent(&self, parent: &WlSurface) -> bool {
+        match &self.shell_surface {
+            ToplevelKind::Xdg(toplevel) => xdg_handlers::is_parent(toplevel, parent),
+            ToplevelKind::ZxdgV6(toplevel) => zxdgv6_handlers::is_parent(toplevel, parent),
+        }
+    }
+
+    /// Sets the parent of this toplevel surface.
+    ///
+    /// The parent must be another toplevel surface.
+    ///
+    /// If the parent is `None`, the parent-child relationship is removed.
+    pub fn set_parent(&self, parent: Option<ToplevelSurface>) {
+        match &self.shell_surface {
+            ToplevelKind::Xdg(toplevel) => {
+                xdg_handlers::set_parent(toplevel, parent.and_then(|parent| parent.get_surface().cloned()))
+            }
+            ToplevelKind::ZxdgV6(toplevel) => {
+                zxdgv6_handlers::set_parent(toplevel, parent.and_then(|parent| parent.get_surface().cloned()))
+            }
+        }
+    }
```

Comment:
In general the parent as set by the compositor is not necessarily always a `{z}xdg_toplevel`, but could potentially be other things (like a layer-shell surface for example), so this method should take a `WlSurface` as argument directly.